moveos/moveos-stdlib/src/natives/moveos_stdlib/raw_table/mod.rs

```diff
@@ -438,6 +490,69 @@ impl CommonGasParameters {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct NewTableGasParameters {
+    pub base: InternalGas,
+    pub per_byte_in_str: InternalGasPerByte,
+}
+
+// native fun new<K: copy + drop>(table_handle: TableHandle): TableInfo;
+fn native_new_table(
+    _common_gas_params: &CommonGasParameters,
+    gas_params: &NewTableGasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    //0 K Type
+    assert_eq!(ty_args.len(), 1);
+    assert_eq!(args.len(), 1);
+
+    let table_context = context.extensions().get::<NativeTableContext>();
+    let mut table_data = table_context.table_data.write();
+
+    let handle = get_table_handle(&mut args)?;
+
+    let mut cost = gas_params.base;
+
+    let table = table_data.get_or_create_table(context, handle, &ty_args[0])?;
+
+    // Table SMT root
+    let state_root = table_context.resolver.resolve_state_root().map_err(|err| {
```

Comment:
new table's state_root should be the place holder hash.

Reply:
fixed